File: library/src/main/java/com/dhc/library/data/cache/ICache.java
package com.dhc.library.data.cache;

/**
 * Created by 邓浩宸 on 2016/11/27.
 */

public interface ICache {
    /**
     * store key , value
     * @param key  must be String
     * @param value  Object
     */
    void put(String key, Object value);

    /**
     * Get the value according to the key
     * @param key   String
     * @return  Object
     */
    Object get(String key);

    /**
     * delete data by key
     * @param key String
     */
    void remove(String key);

    /**
     * Contains the value
     * @param key  String
     * @return   Whether or not
    Contains the value
     */
    boolean contains(String key);

    /**
     * Clean all
     */
    void clear();

}

File: library/src/main/java/com/dhc/library/data/cache/ICache.kt
package com.dhc.library.data.cache

/**
 * Created by 邓浩宸 on 2016/11/27.
 */

interface ICache {
    /**
     * store key , value
     * @param key  must be String
     * @param value  Object
     */
    fun put(key: String, value: Any)

    /**
     * Get the value according to the key
     * @param key   String
     * @return  Object
     */
    operator fun get(key: String): Any?

    /**
     * delete data by key
     * @param key String
     */
    fun remove(key: String)

    /**
     * Contains the value
     * @param key  String
     * @return   Whether or not
     * Contains the value
     */
    operator fun contains(key: String): Boolean

    /**
     * Clean all
     */
    fun clear()

}
